telemetry/csharp/AwsToolkit.Telemetry.Events.Tests/GeneratedCodeTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Amazon.AwsToolkit.Telemetry.Events.Core;
+using Amazon.AwsToolkit.Telemetry.Events.Tests.Generated;
+using Moq;
+using Xunit;
+
+namespace Amazon.AwsToolkit.Telemetry.Events.Tests
+{
+    /// <summary>
+    /// Test a sampling of the generated code, to see that RecordXxx calls function as expected
+    /// </summary>
+    public class GeneratedCodeTests
+    {
+        private readonly Mock<ITelemetryLogger> _telemetryLogger = new Mock<ITelemetryLogger>();
+        private Metrics _recordedMetrics = null;
+
+        public GeneratedCodeTests()
+        {
+            _telemetryLogger.Setup(mock => mock.Record(It.IsAny<Metrics>()))
+                .Callback<Metrics>(metrics =>
+                {
+                    _recordedMetrics = metrics;
+                });
+        }
+
+        /// <summary>
+        /// RecordLambdaInvokeRemote was chosen as a sample call that uses a telemetry "enum" type (Runtime)
+        /// </summary>
+        [Fact]
+        public void RecordLambdaInvokeRemote()
```

Comment:
Do you see this as a test to prove out telemetry as a whole (eg. if these pass we assume all generated code will pass)? Or would we need to change these as we add new metrics?

Reply:
In an ideal world, we'd auto generate code that tests each of the generated methods.

Because all of the RecordXxx call implementations are predictable, I think its reasonable to pick a sampling of calls that exercise the different handling scenarios/types. I don't see us adding tests for every new metric that is added.